### PR TITLE
Added security CI workflow and OS build qualifier

### PR DESCRIPTION
**Pull request by saikaranam-amazon**
Signed-off-by: Sai Kumar <karanas@amazon.com>

### Description
Added security CI workflow and OS build qualifier
 
### Issues Resolved
https://github.com/opensearch-project/cross-cluster-replication/issues/354
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]
build.gradle 83-86 should change to
`version = "${opensearch_build}"

File: .github/workflows/security-tests.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        uses: actions/checkout@v2
+        with:
+          repository: 'opensearch-project/security'
+          path: security
+          ref: 'main'
+      - name: Build security
+        working-directory: ./security
+        run: |
+          ./gradlew clean build -Dbuild.snapshot=false -x test
+          cp build/distributions/opensearch-security-*.zip ../src/test/resources/security/plugin/opensearch-security.zip
```
[the remark this answers is not in the document]
Comment (saikaranam-amazon) [reply .github/workflows/security-tests.yml:36]: Tracked here: https://github.com/opensearch-project/cross-cluster-replication/issues/274

**Comment by saikaranam-amazon** [comment]
> build.gradle 83-86 should change to
version = "${opensearch_build}"

Done

**Review by naveenpajjuri — APPROVED** [review]

**Review by ankitkala — APPROVED** [review]
LGTM.

**Pull request merged**